lib/default.nix

```diff
@@ -148,6 +148,52 @@ let
           _module.args.perSystem = systemArgs.${pkgs.system}.perSystem;
         };
 
+      homes =
+        let
+          loadHomeConfiguration =
+            path: pkgs:
+            inputs.home-manager.lib.homeManagerConfiguration {
+              inherit pkgs;
+              extraSpecialArgs = specialArgs;
+              modules = [
+                perSystemModule
+                path
+              ];
+            };
+
+          loadDefault = path: pkgs: (import path) { inherit pkgs inputs flake; };
+
+          mkUserConfig =
+            hostname: username: entry: pkgs:
+            if entry.type == "regular" then
+              loadHomeConfiguration entry.path pkgs
+            else if builtins.pathExists (entry.path + "/default.nix") then
+              loadDefault (entry.path + "/default.nix") pkgs
+            else if builtins.pathExists (entry.path + "/home-configuration.nix") then
+              loadHomeConfiguration (entry.path + "/home-configuration.nix") pkgs
+            else
+              throw "home '${username}@${hostname}' does not have a configuration";
+
+          mkHostUsers =
+            hostname: entries:
+            let
+              validUsers = lib.filterAttrs (_: { type, ... }: type == "regular" || type == "directory") entries;
+            in
+            lib.concatMapAttrs (username: entry: {
+              "${username}@${hostname}" = mkUserConfig hostname username entry;
+            }) validUsers;
```

Comment:
One aspect, I believe as a new user, it would be surprising to me if I had to plug the home-manager into the NixOS config manually.

Reply:
Fully agree - and it feels good in practice too.